packages/kbn-esql-validation-autocomplete/src/definitions/types.ts

```diff
@@ -8,21 +8,30 @@
 
 import type { ESQLCommand, ESQLCommandOption, ESQLFunction, ESQLMessage } from '@kbn/esql-ast';
 
+export const supportedFieldTypes = [
+  'number',
+  'date',
+  'string',
+  'boolean',
+  'ip',
+  'cartesian_point',
+  'cartesian_shape',
+  'geo_point',
+  'geo_shape',
+  'version',
+] as const;
+
+export const isSupportedFieldType = (type: string): type is SupportedFieldType =>
+  supportedFieldTypes.includes(type as SupportedFieldType);
+
+export type SupportedFieldType = typeof supportedFieldTypes[number];
```

Comment:
This typing is very powerful and will be extremely helpful for https://github.com/elastic/kibana/issues/174710 since it will prompt us to update the validation tests and the generator in the appropriate places